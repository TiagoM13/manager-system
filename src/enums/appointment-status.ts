export enum AppointmentStatus {
  CONPLETED = 'conclu√≠da',
  CANCELLED = 'cancelada',
  PENDING = 'pendente',
}
