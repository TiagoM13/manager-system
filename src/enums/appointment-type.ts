export enum AppointmentType {
  QUERY = 'consulta',
  OTHER = 'outros',
  ACCIDENT = 'acidente de tr√¢nsito',
  FIREARM_INJURY = 'ferimento por arma de fogo',
  WHITE_WEAPON_INJURY = 'ferimento por arma branca',
}
