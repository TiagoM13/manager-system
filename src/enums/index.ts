export * from './role';
export * from './status';
export * from './appointment-type';
export * from './appointment-status';
