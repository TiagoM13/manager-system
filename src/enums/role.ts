export enum Role {
  ADMIN = 'administrador',
  EDITOR = 'editor',
  CLINICAL = 'cl√≠nico',
}
