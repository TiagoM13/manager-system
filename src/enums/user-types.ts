export enum UserTypes {
  ADMIN = 'administrador',
  EDITOR = 'editor',
  CLINICAL = 'clínico',
  NO_TYPE = 'não definido'
}
