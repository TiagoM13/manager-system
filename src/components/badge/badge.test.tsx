import React from 'react';

import { Status, UserTypes } from '@/enums';
import { render } from '@testing-library/react';

import { Badge } from '.';

describe('<Badge />', () => {
  test('should render correctly component', () => {
    const { getByTestId, getByText } = render(<Badge type={UserTypes.ADMIN} />);

    const container = getByTestId('badge-container');

    expect(getByText(UserTypes.ADMIN)).toBeTruthy();
    expect(container).toHaveClass('bg-purple-500');
  });

  test('should render correctly className', () => {
    const { getByTestId, getByText } = render(
      <Badge type={UserTypes.EDITOR} />,
    );

    const container = getByTestId('badge-container');

    expect(getByText(UserTypes.EDITOR)).toBeTruthy();
    expect(container).toHaveClass('bg-blue-500');
    expect(container).not.toHaveClass('bg-purple-500');
  });

  test('should render correctly status active', () => {
    const { getByTestId, getByText, queryAllByText } = render(
      <Badge type={Status.ACTIVE} />,
    );

    const container = getByTestId('badge-container');

    expect(getByText(Status.ACTIVE)).toBeTruthy();
    expect(queryAllByText(Status.INACTIVE).length).not.toBeTruthy();
    expect(container).toHaveClass('bg-emerald-400');
    expect(container).not.toHaveClass('bg-slate-300');
  });
});
