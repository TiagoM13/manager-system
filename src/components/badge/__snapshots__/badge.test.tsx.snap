// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render correctly status active 1`] = `
<DocumentFragment>
  <div
    class="bg-emerald-400 text-black font-medium rounded-xl text-center text-xs px-3 py-1.5 w-fit"
    data-testid="badge-container"
  >
    <span
      class="capitalize"
    >
      ativo
    </span>
  </div>
</DocumentFragment>
`;
